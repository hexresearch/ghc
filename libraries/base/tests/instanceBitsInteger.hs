module Main where

import Control.Monad (when)
import Data.Bits
import Data.Int (Int32, Int64)
import Data.Foldable (for_)
import GHC.Num.Integer (integerCheck)

integers :: [Integer]
integers = concatMap f [minInt64, minInt32, 0, maxInt32, maxInt64]
  where
    f i = [pred $ pred i, pred i, i, succ i, succ $ succ i]
    minInt64 = toInteger (minBound :: Int64)
    minInt32 = toInteger (minBound :: Int32)
    maxInt32 = toInteger (maxBound :: Int32)
    maxInt64 = toInteger (maxBound :: Int64)

bits :: [Int]
bits = [0,1,62,63,64]

testXBit name f model = do
  putStrLn name
  for_ integers $ \i ->
    for_ bits $ \b -> do
      let actual = f i b
      let expected = model i b
      putStr $ name ++ " (" ++ show i ++ ") " ++ show b ++ " = " ++ show actual
      putStr ", "
      putStr $ if integerCheck actual then "valid Integer" else "invalid Integer"
      putStr ", "
      putStrLn $ if actual == expected then "matches default implementation" else "doesn't match default implementation"
  putStrLn ""

main :: IO ()
main = do
  testXBit "setBit" setBit (\i b -> i .|. bit b)
  testXBit "clearBit" clearBit (\i b -> i .&. complement (bit b))
  testXBit "complementBit" complementBit (\i b -> i `xor` bit b)
