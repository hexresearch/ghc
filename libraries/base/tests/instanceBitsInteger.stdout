setBit
setBit (-9223372036854775810) 0 = -9223372036854775809, valid Integer, matches default implementation
setBit (-9223372036854775810) 1 = -9223372036854775810, valid Integer, matches default implementation
setBit (-9223372036854775810) 62 = -9223372036854775810, valid Integer, matches default implementation
setBit (-9223372036854775810) 63 = -2, valid Integer, matches default implementation
setBit (-9223372036854775810) 64 = -9223372036854775810, valid Integer, matches default implementation
setBit (-9223372036854775809) 0 = -9223372036854775809, valid Integer, matches default implementation
setBit (-9223372036854775809) 1 = -9223372036854775809, valid Integer, matches default implementation
setBit (-9223372036854775809) 62 = -9223372036854775809, valid Integer, matches default implementation
setBit (-9223372036854775809) 63 = -1, valid Integer, matches default implementation
setBit (-9223372036854775809) 64 = -9223372036854775809, valid Integer, matches default implementation
setBit (-9223372036854775808) 0 = -9223372036854775807, valid Integer, matches default implementation
setBit (-9223372036854775808) 1 = -9223372036854775806, valid Integer, matches default implementation
setBit (-9223372036854775808) 62 = -4611686018427387904, valid Integer, matches default implementation
setBit (-9223372036854775808) 63 = -9223372036854775808, valid Integer, matches default implementation
setBit (-9223372036854775808) 64 = -9223372036854775808, valid Integer, matches default implementation
setBit (-9223372036854775807) 0 = -9223372036854775807, valid Integer, matches default implementation
setBit (-9223372036854775807) 1 = -9223372036854775805, valid Integer, matches default implementation
setBit (-9223372036854775807) 62 = -4611686018427387903, valid Integer, matches default implementation
setBit (-9223372036854775807) 63 = -9223372036854775807, valid Integer, matches default implementation
setBit (-9223372036854775807) 64 = -9223372036854775807, valid Integer, matches default implementation
setBit (-9223372036854775806) 0 = -9223372036854775805, valid Integer, matches default implementation
setBit (-9223372036854775806) 1 = -9223372036854775806, valid Integer, matches default implementation
setBit (-9223372036854775806) 62 = -4611686018427387902, valid Integer, matches default implementation
setBit (-9223372036854775806) 63 = -9223372036854775806, valid Integer, matches default implementation
setBit (-9223372036854775806) 64 = -9223372036854775806, valid Integer, matches default implementation
setBit (-2147483650) 0 = -2147483649, valid Integer, matches default implementation
setBit (-2147483650) 1 = -2147483650, valid Integer, matches default implementation
setBit (-2147483650) 62 = -2147483650, valid Integer, matches default implementation
setBit (-2147483650) 63 = -2147483650, valid Integer, matches default implementation
setBit (-2147483650) 64 = -2147483650, valid Integer, matches default implementation
setBit (-2147483649) 0 = -2147483649, valid Integer, matches default implementation
setBit (-2147483649) 1 = -2147483649, valid Integer, matches default implementation
setBit (-2147483649) 62 = -2147483649, valid Integer, matches default implementation
setBit (-2147483649) 63 = -2147483649, valid Integer, matches default implementation
setBit (-2147483649) 64 = -2147483649, valid Integer, matches default implementation
setBit (-2147483648) 0 = -2147483647, valid Integer, matches default implementation
setBit (-2147483648) 1 = -2147483646, valid Integer, matches default implementation
setBit (-2147483648) 62 = -2147483648, valid Integer, matches default implementation
setBit (-2147483648) 63 = -2147483648, valid Integer, matches default implementation
setBit (-2147483648) 64 = -2147483648, valid Integer, matches default implementation
setBit (-2147483647) 0 = -2147483647, valid Integer, matches default implementation
setBit (-2147483647) 1 = -2147483645, valid Integer, matches default implementation
setBit (-2147483647) 62 = -2147483647, valid Integer, matches default implementation
setBit (-2147483647) 63 = -2147483647, valid Integer, matches default implementation
setBit (-2147483647) 64 = -2147483647, valid Integer, matches default implementation
setBit (-2147483646) 0 = -2147483645, valid Integer, matches default implementation
setBit (-2147483646) 1 = -2147483646, valid Integer, matches default implementation
setBit (-2147483646) 62 = -2147483646, valid Integer, matches default implementation
setBit (-2147483646) 63 = -2147483646, valid Integer, matches default implementation
setBit (-2147483646) 64 = -2147483646, valid Integer, matches default implementation
setBit (-2) 0 = -1, valid Integer, matches default implementation
setBit (-2) 1 = -2, valid Integer, matches default implementation
setBit (-2) 62 = -2, valid Integer, matches default implementation
setBit (-2) 63 = -2, valid Integer, matches default implementation
setBit (-2) 64 = -2, valid Integer, matches default implementation
setBit (-1) 0 = -1, valid Integer, matches default implementation
setBit (-1) 1 = -1, valid Integer, matches default implementation
setBit (-1) 62 = -1, valid Integer, matches default implementation
setBit (-1) 63 = -1, valid Integer, matches default implementation
setBit (-1) 64 = -1, valid Integer, matches default implementation
setBit (0) 0 = 1, valid Integer, matches default implementation
setBit (0) 1 = 2, valid Integer, matches default implementation
setBit (0) 62 = 4611686018427387904, valid Integer, matches default implementation
setBit (0) 63 = 9223372036854775808, valid Integer, matches default implementation
setBit (0) 64 = 18446744073709551616, valid Integer, matches default implementation
setBit (1) 0 = 1, valid Integer, matches default implementation
setBit (1) 1 = 3, valid Integer, matches default implementation
setBit (1) 62 = 4611686018427387905, valid Integer, matches default implementation
setBit (1) 63 = 9223372036854775809, valid Integer, matches default implementation
setBit (1) 64 = 18446744073709551617, valid Integer, matches default implementation
setBit (2) 0 = 3, valid Integer, matches default implementation
setBit (2) 1 = 2, valid Integer, matches default implementation
setBit (2) 62 = 4611686018427387906, valid Integer, matches default implementation
setBit (2) 63 = 9223372036854775810, valid Integer, matches default implementation
setBit (2) 64 = 18446744073709551618, valid Integer, matches default implementation
setBit (2147483645) 0 = 2147483645, valid Integer, matches default implementation
setBit (2147483645) 1 = 2147483647, valid Integer, matches default implementation
setBit (2147483645) 62 = 4611686020574871549, valid Integer, matches default implementation
setBit (2147483645) 63 = 9223372039002259453, valid Integer, matches default implementation
setBit (2147483645) 64 = 18446744075857035261, valid Integer, matches default implementation
setBit (2147483646) 0 = 2147483647, valid Integer, matches default implementation
setBit (2147483646) 1 = 2147483646, valid Integer, matches default implementation
setBit (2147483646) 62 = 4611686020574871550, valid Integer, matches default implementation
setBit (2147483646) 63 = 9223372039002259454, valid Integer, matches default implementation
setBit (2147483646) 64 = 18446744075857035262, valid Integer, matches default implementation
setBit (2147483647) 0 = 2147483647, valid Integer, matches default implementation
setBit (2147483647) 1 = 2147483647, valid Integer, matches default implementation
setBit (2147483647) 62 = 4611686020574871551, valid Integer, matches default implementation
setBit (2147483647) 63 = 9223372039002259455, valid Integer, matches default implementation
setBit (2147483647) 64 = 18446744075857035263, valid Integer, matches default implementation
setBit (2147483648) 0 = 2147483649, valid Integer, matches default implementation
setBit (2147483648) 1 = 2147483650, valid Integer, matches default implementation
setBit (2147483648) 62 = 4611686020574871552, valid Integer, matches default implementation
setBit (2147483648) 63 = 9223372039002259456, valid Integer, matches default implementation
setBit (2147483648) 64 = 18446744075857035264, valid Integer, matches default implementation
setBit (2147483649) 0 = 2147483649, valid Integer, matches default implementation
setBit (2147483649) 1 = 2147483651, valid Integer, matches default implementation
setBit (2147483649) 62 = 4611686020574871553, valid Integer, matches default implementation
setBit (2147483649) 63 = 9223372039002259457, valid Integer, matches default implementation
setBit (2147483649) 64 = 18446744075857035265, valid Integer, matches default implementation
setBit (9223372036854775805) 0 = 9223372036854775805, valid Integer, matches default implementation
setBit (9223372036854775805) 1 = 9223372036854775807, valid Integer, matches default implementation
setBit (9223372036854775805) 62 = 9223372036854775805, valid Integer, matches default implementation
setBit (9223372036854775805) 63 = 18446744073709551613, valid Integer, matches default implementation
setBit (9223372036854775805) 64 = 27670116110564327421, valid Integer, matches default implementation
setBit (9223372036854775806) 0 = 9223372036854775807, valid Integer, matches default implementation
setBit (9223372036854775806) 1 = 9223372036854775806, valid Integer, matches default implementation
setBit (9223372036854775806) 62 = 9223372036854775806, valid Integer, matches default implementation
setBit (9223372036854775806) 63 = 18446744073709551614, valid Integer, matches default implementation
setBit (9223372036854775806) 64 = 27670116110564327422, valid Integer, matches default implementation
setBit (9223372036854775807) 0 = 9223372036854775807, valid Integer, matches default implementation
setBit (9223372036854775807) 1 = 9223372036854775807, valid Integer, matches default implementation
setBit (9223372036854775807) 62 = 9223372036854775807, valid Integer, matches default implementation
setBit (9223372036854775807) 63 = 18446744073709551615, valid Integer, matches default implementation
setBit (9223372036854775807) 64 = 27670116110564327423, valid Integer, matches default implementation
setBit (9223372036854775808) 0 = 9223372036854775809, valid Integer, matches default implementation
setBit (9223372036854775808) 1 = 9223372036854775810, valid Integer, matches default implementation
setBit (9223372036854775808) 62 = 13835058055282163712, valid Integer, matches default implementation
setBit (9223372036854775808) 63 = 9223372036854775808, valid Integer, matches default implementation
setBit (9223372036854775808) 64 = 27670116110564327424, valid Integer, matches default implementation
setBit (9223372036854775809) 0 = 9223372036854775809, valid Integer, matches default implementation
setBit (9223372036854775809) 1 = 9223372036854775811, valid Integer, matches default implementation
setBit (9223372036854775809) 62 = 13835058055282163713, valid Integer, matches default implementation
setBit (9223372036854775809) 63 = 9223372036854775809, valid Integer, matches default implementation
setBit (9223372036854775809) 64 = 27670116110564327425, valid Integer, matches default implementation

clearBit
clearBit (-9223372036854775810) 0 = -9223372036854775810, valid Integer, matches default implementation
clearBit (-9223372036854775810) 1 = -9223372036854775812, valid Integer, matches default implementation
clearBit (-9223372036854775810) 62 = -13835058055282163714, valid Integer, matches default implementation
clearBit (-9223372036854775810) 63 = -9223372036854775810, valid Integer, matches default implementation
clearBit (-9223372036854775810) 64 = -27670116110564327426, valid Integer, matches default implementation
clearBit (-9223372036854775809) 0 = -9223372036854775810, valid Integer, matches default implementation
clearBit (-9223372036854775809) 1 = -9223372036854775811, valid Integer, matches default implementation
clearBit (-9223372036854775809) 62 = -13835058055282163713, valid Integer, matches default implementation
clearBit (-9223372036854775809) 63 = -9223372036854775809, valid Integer, matches default implementation
clearBit (-9223372036854775809) 64 = -27670116110564327425, valid Integer, matches default implementation
clearBit (-9223372036854775808) 0 = -9223372036854775808, valid Integer, matches default implementation
clearBit (-9223372036854775808) 1 = -9223372036854775808, valid Integer, matches default implementation
clearBit (-9223372036854775808) 62 = -9223372036854775808, valid Integer, matches default implementation
clearBit (-9223372036854775808) 63 = -18446744073709551616, valid Integer, matches default implementation
clearBit (-9223372036854775808) 64 = -27670116110564327424, valid Integer, matches default implementation
clearBit (-9223372036854775807) 0 = -9223372036854775808, valid Integer, matches default implementation
clearBit (-9223372036854775807) 1 = -9223372036854775807, valid Integer, matches default implementation
clearBit (-9223372036854775807) 62 = -9223372036854775807, valid Integer, matches default implementation
clearBit (-9223372036854775807) 63 = -18446744073709551615, valid Integer, matches default implementation
clearBit (-9223372036854775807) 64 = -27670116110564327423, valid Integer, matches default implementation
clearBit (-9223372036854775806) 0 = -9223372036854775806, valid Integer, matches default implementation
clearBit (-9223372036854775806) 1 = -9223372036854775808, valid Integer, matches default implementation
clearBit (-9223372036854775806) 62 = -9223372036854775806, valid Integer, matches default implementation
clearBit (-9223372036854775806) 63 = -18446744073709551614, valid Integer, matches default implementation
clearBit (-9223372036854775806) 64 = -27670116110564327422, valid Integer, matches default implementation
clearBit (-2147483650) 0 = -2147483650, valid Integer, matches default implementation
clearBit (-2147483650) 1 = -2147483652, valid Integer, matches default implementation
clearBit (-2147483650) 62 = -4611686020574871554, valid Integer, matches default implementation
clearBit (-2147483650) 63 = -9223372039002259458, valid Integer, matches default implementation
clearBit (-2147483650) 64 = -18446744075857035266, valid Integer, matches default implementation
clearBit (-2147483649) 0 = -2147483650, valid Integer, matches default implementation
clearBit (-2147483649) 1 = -2147483651, valid Integer, matches default implementation
clearBit (-2147483649) 62 = -4611686020574871553, valid Integer, matches default implementation
clearBit (-2147483649) 63 = -9223372039002259457, valid Integer, matches default implementation
clearBit (-2147483649) 64 = -18446744075857035265, valid Integer, matches default implementation
clearBit (-2147483648) 0 = -2147483648, valid Integer, matches default implementation
clearBit (-2147483648) 1 = -2147483648, valid Integer, matches default implementation
clearBit (-2147483648) 62 = -4611686020574871552, valid Integer, matches default implementation
clearBit (-2147483648) 63 = -9223372039002259456, valid Integer, matches default implementation
clearBit (-2147483648) 64 = -18446744075857035264, valid Integer, matches default implementation
clearBit (-2147483647) 0 = -2147483648, valid Integer, matches default implementation
clearBit (-2147483647) 1 = -2147483647, valid Integer, matches default implementation
clearBit (-2147483647) 62 = -4611686020574871551, valid Integer, matches default implementation
clearBit (-2147483647) 63 = -9223372039002259455, valid Integer, matches default implementation
clearBit (-2147483647) 64 = -18446744075857035263, valid Integer, matches default implementation
clearBit (-2147483646) 0 = -2147483646, valid Integer, matches default implementation
clearBit (-2147483646) 1 = -2147483648, valid Integer, matches default implementation
clearBit (-2147483646) 62 = -4611686020574871550, valid Integer, matches default implementation
clearBit (-2147483646) 63 = -9223372039002259454, valid Integer, matches default implementation
clearBit (-2147483646) 64 = -18446744075857035262, valid Integer, matches default implementation
clearBit (-2) 0 = -2, valid Integer, matches default implementation
clearBit (-2) 1 = -4, valid Integer, matches default implementation
clearBit (-2) 62 = -4611686018427387906, valid Integer, matches default implementation
clearBit (-2) 63 = -9223372036854775810, valid Integer, matches default implementation
clearBit (-2) 64 = -18446744073709551618, valid Integer, matches default implementation
clearBit (-1) 0 = -2, valid Integer, matches default implementation
clearBit (-1) 1 = -3, valid Integer, matches default implementation
clearBit (-1) 62 = -4611686018427387905, valid Integer, matches default implementation
clearBit (-1) 63 = -9223372036854775809, valid Integer, matches default implementation
clearBit (-1) 64 = -18446744073709551617, valid Integer, matches default implementation
clearBit (0) 0 = 0, valid Integer, matches default implementation
clearBit (0) 1 = 0, valid Integer, matches default implementation
clearBit (0) 62 = 0, valid Integer, matches default implementation
clearBit (0) 63 = 0, valid Integer, matches default implementation
clearBit (0) 64 = 0, valid Integer, matches default implementation
clearBit (1) 0 = 0, valid Integer, matches default implementation
clearBit (1) 1 = 1, valid Integer, matches default implementation
clearBit (1) 62 = 1, valid Integer, matches default implementation
clearBit (1) 63 = 1, valid Integer, matches default implementation
clearBit (1) 64 = 1, valid Integer, matches default implementation
clearBit (2) 0 = 2, valid Integer, matches default implementation
clearBit (2) 1 = 0, valid Integer, matches default implementation
clearBit (2) 62 = 2, valid Integer, matches default implementation
clearBit (2) 63 = 2, valid Integer, matches default implementation
clearBit (2) 64 = 2, valid Integer, matches default implementation
clearBit (2147483645) 0 = 2147483644, valid Integer, matches default implementation
clearBit (2147483645) 1 = 2147483645, valid Integer, matches default implementation
clearBit (2147483645) 62 = 2147483645, valid Integer, matches default implementation
clearBit (2147483645) 63 = 2147483645, valid Integer, matches default implementation
clearBit (2147483645) 64 = 2147483645, valid Integer, matches default implementation
clearBit (2147483646) 0 = 2147483646, valid Integer, matches default implementation
clearBit (2147483646) 1 = 2147483644, valid Integer, matches default implementation
clearBit (2147483646) 62 = 2147483646, valid Integer, matches default implementation
clearBit (2147483646) 63 = 2147483646, valid Integer, matches default implementation
clearBit (2147483646) 64 = 2147483646, valid Integer, matches default implementation
clearBit (2147483647) 0 = 2147483646, valid Integer, matches default implementation
clearBit (2147483647) 1 = 2147483645, valid Integer, matches default implementation
clearBit (2147483647) 62 = 2147483647, valid Integer, matches default implementation
clearBit (2147483647) 63 = 2147483647, valid Integer, matches default implementation
clearBit (2147483647) 64 = 2147483647, valid Integer, matches default implementation
clearBit (2147483648) 0 = 2147483648, valid Integer, matches default implementation
clearBit (2147483648) 1 = 2147483648, valid Integer, matches default implementation
clearBit (2147483648) 62 = 2147483648, valid Integer, matches default implementation
clearBit (2147483648) 63 = 2147483648, valid Integer, matches default implementation
clearBit (2147483648) 64 = 2147483648, valid Integer, matches default implementation
clearBit (2147483649) 0 = 2147483648, valid Integer, matches default implementation
clearBit (2147483649) 1 = 2147483649, valid Integer, matches default implementation
clearBit (2147483649) 62 = 2147483649, valid Integer, matches default implementation
clearBit (2147483649) 63 = 2147483649, valid Integer, matches default implementation
clearBit (2147483649) 64 = 2147483649, valid Integer, matches default implementation
clearBit (9223372036854775805) 0 = 9223372036854775804, valid Integer, matches default implementation
clearBit (9223372036854775805) 1 = 9223372036854775805, valid Integer, matches default implementation
clearBit (9223372036854775805) 62 = 4611686018427387901, valid Integer, matches default implementation
clearBit (9223372036854775805) 63 = 9223372036854775805, valid Integer, matches default implementation
clearBit (9223372036854775805) 64 = 9223372036854775805, valid Integer, matches default implementation
clearBit (9223372036854775806) 0 = 9223372036854775806, valid Integer, matches default implementation
clearBit (9223372036854775806) 1 = 9223372036854775804, valid Integer, matches default implementation
clearBit (9223372036854775806) 62 = 4611686018427387902, valid Integer, matches default implementation
clearBit (9223372036854775806) 63 = 9223372036854775806, valid Integer, matches default implementation
clearBit (9223372036854775806) 64 = 9223372036854775806, valid Integer, matches default implementation
clearBit (9223372036854775807) 0 = 9223372036854775806, valid Integer, matches default implementation
clearBit (9223372036854775807) 1 = 9223372036854775805, valid Integer, matches default implementation
clearBit (9223372036854775807) 62 = 4611686018427387903, valid Integer, matches default implementation
clearBit (9223372036854775807) 63 = 9223372036854775807, valid Integer, matches default implementation
clearBit (9223372036854775807) 64 = 9223372036854775807, valid Integer, matches default implementation
clearBit (9223372036854775808) 0 = 9223372036854775808, valid Integer, matches default implementation
clearBit (9223372036854775808) 1 = 9223372036854775808, valid Integer, matches default implementation
clearBit (9223372036854775808) 62 = 9223372036854775808, valid Integer, matches default implementation
clearBit (9223372036854775808) 63 = 0, valid Integer, matches default implementation
clearBit (9223372036854775808) 64 = 9223372036854775808, valid Integer, matches default implementation
clearBit (9223372036854775809) 0 = 9223372036854775808, valid Integer, matches default implementation
clearBit (9223372036854775809) 1 = 9223372036854775809, valid Integer, matches default implementation
clearBit (9223372036854775809) 62 = 9223372036854775809, valid Integer, matches default implementation
clearBit (9223372036854775809) 63 = 1, valid Integer, matches default implementation
clearBit (9223372036854775809) 64 = 9223372036854775809, valid Integer, matches default implementation

complementBit
complementBit (-9223372036854775810) 0 = -9223372036854775809, valid Integer, matches default implementation
complementBit (-9223372036854775810) 1 = -9223372036854775812, valid Integer, matches default implementation
complementBit (-9223372036854775810) 62 = -13835058055282163714, valid Integer, matches default implementation
complementBit (-9223372036854775810) 63 = -2, valid Integer, matches default implementation
complementBit (-9223372036854775810) 64 = -27670116110564327426, valid Integer, matches default implementation
complementBit (-9223372036854775809) 0 = -9223372036854775810, valid Integer, matches default implementation
complementBit (-9223372036854775809) 1 = -9223372036854775811, valid Integer, matches default implementation
complementBit (-9223372036854775809) 62 = -13835058055282163713, valid Integer, matches default implementation
complementBit (-9223372036854775809) 63 = -1, valid Integer, matches default implementation
complementBit (-9223372036854775809) 64 = -27670116110564327425, valid Integer, matches default implementation
complementBit (-9223372036854775808) 0 = -9223372036854775807, valid Integer, matches default implementation
complementBit (-9223372036854775808) 1 = -9223372036854775806, valid Integer, matches default implementation
complementBit (-9223372036854775808) 62 = -4611686018427387904, valid Integer, matches default implementation
complementBit (-9223372036854775808) 63 = -18446744073709551616, valid Integer, matches default implementation
complementBit (-9223372036854775808) 64 = -27670116110564327424, valid Integer, matches default implementation
complementBit (-9223372036854775807) 0 = -9223372036854775808, valid Integer, matches default implementation
complementBit (-9223372036854775807) 1 = -9223372036854775805, valid Integer, matches default implementation
complementBit (-9223372036854775807) 62 = -4611686018427387903, valid Integer, matches default implementation
complementBit (-9223372036854775807) 63 = -18446744073709551615, valid Integer, matches default implementation
complementBit (-9223372036854775807) 64 = -27670116110564327423, valid Integer, matches default implementation
complementBit (-9223372036854775806) 0 = -9223372036854775805, valid Integer, matches default implementation
complementBit (-9223372036854775806) 1 = -9223372036854775808, valid Integer, matches default implementation
complementBit (-9223372036854775806) 62 = -4611686018427387902, valid Integer, matches default implementation
complementBit (-9223372036854775806) 63 = -18446744073709551614, valid Integer, matches default implementation
complementBit (-9223372036854775806) 64 = -27670116110564327422, valid Integer, matches default implementation
complementBit (-2147483650) 0 = -2147483649, valid Integer, matches default implementation
complementBit (-2147483650) 1 = -2147483652, valid Integer, matches default implementation
complementBit (-2147483650) 62 = -4611686020574871554, valid Integer, matches default implementation
complementBit (-2147483650) 63 = -9223372039002259458, valid Integer, matches default implementation
complementBit (-2147483650) 64 = -18446744075857035266, valid Integer, matches default implementation
complementBit (-2147483649) 0 = -2147483650, valid Integer, matches default implementation
complementBit (-2147483649) 1 = -2147483651, valid Integer, matches default implementation
complementBit (-2147483649) 62 = -4611686020574871553, valid Integer, matches default implementation
complementBit (-2147483649) 63 = -9223372039002259457, valid Integer, matches default implementation
complementBit (-2147483649) 64 = -18446744075857035265, valid Integer, matches default implementation
complementBit (-2147483648) 0 = -2147483647, valid Integer, matches default implementation
complementBit (-2147483648) 1 = -2147483646, valid Integer, matches default implementation
complementBit (-2147483648) 62 = -4611686020574871552, valid Integer, matches default implementation
complementBit (-2147483648) 63 = -9223372039002259456, valid Integer, matches default implementation
complementBit (-2147483648) 64 = -18446744075857035264, valid Integer, matches default implementation
complementBit (-2147483647) 0 = -2147483648, valid Integer, matches default implementation
complementBit (-2147483647) 1 = -2147483645, valid Integer, matches default implementation
complementBit (-2147483647) 62 = -4611686020574871551, valid Integer, matches default implementation
complementBit (-2147483647) 63 = -9223372039002259455, valid Integer, matches default implementation
complementBit (-2147483647) 64 = -18446744075857035263, valid Integer, matches default implementation
complementBit (-2147483646) 0 = -2147483645, valid Integer, matches default implementation
complementBit (-2147483646) 1 = -2147483648, valid Integer, matches default implementation
complementBit (-2147483646) 62 = -4611686020574871550, valid Integer, matches default implementation
complementBit (-2147483646) 63 = -9223372039002259454, valid Integer, matches default implementation
complementBit (-2147483646) 64 = -18446744075857035262, valid Integer, matches default implementation
complementBit (-2) 0 = -1, valid Integer, matches default implementation
complementBit (-2) 1 = -4, valid Integer, matches default implementation
complementBit (-2) 62 = -4611686018427387906, valid Integer, matches default implementation
complementBit (-2) 63 = -9223372036854775810, valid Integer, matches default implementation
complementBit (-2) 64 = -18446744073709551618, valid Integer, matches default implementation
complementBit (-1) 0 = -2, valid Integer, matches default implementation
complementBit (-1) 1 = -3, valid Integer, matches default implementation
complementBit (-1) 62 = -4611686018427387905, valid Integer, matches default implementation
complementBit (-1) 63 = -9223372036854775809, valid Integer, matches default implementation
complementBit (-1) 64 = -18446744073709551617, valid Integer, matches default implementation
complementBit (0) 0 = 1, valid Integer, matches default implementation
complementBit (0) 1 = 2, valid Integer, matches default implementation
complementBit (0) 62 = 4611686018427387904, valid Integer, matches default implementation
complementBit (0) 63 = 9223372036854775808, valid Integer, matches default implementation
complementBit (0) 64 = 18446744073709551616, valid Integer, matches default implementation
complementBit (1) 0 = 0, valid Integer, matches default implementation
complementBit (1) 1 = 3, valid Integer, matches default implementation
complementBit (1) 62 = 4611686018427387905, valid Integer, matches default implementation
complementBit (1) 63 = 9223372036854775809, valid Integer, matches default implementation
complementBit (1) 64 = 18446744073709551617, valid Integer, matches default implementation
complementBit (2) 0 = 3, valid Integer, matches default implementation
complementBit (2) 1 = 0, valid Integer, matches default implementation
complementBit (2) 62 = 4611686018427387906, valid Integer, matches default implementation
complementBit (2) 63 = 9223372036854775810, valid Integer, matches default implementation
complementBit (2) 64 = 18446744073709551618, valid Integer, matches default implementation
complementBit (2147483645) 0 = 2147483644, valid Integer, matches default implementation
complementBit (2147483645) 1 = 2147483647, valid Integer, matches default implementation
complementBit (2147483645) 62 = 4611686020574871549, valid Integer, matches default implementation
complementBit (2147483645) 63 = 9223372039002259453, valid Integer, matches default implementation
complementBit (2147483645) 64 = 18446744075857035261, valid Integer, matches default implementation
complementBit (2147483646) 0 = 2147483647, valid Integer, matches default implementation
complementBit (2147483646) 1 = 2147483644, valid Integer, matches default implementation
complementBit (2147483646) 62 = 4611686020574871550, valid Integer, matches default implementation
complementBit (2147483646) 63 = 9223372039002259454, valid Integer, matches default implementation
complementBit (2147483646) 64 = 18446744075857035262, valid Integer, matches default implementation
complementBit (2147483647) 0 = 2147483646, valid Integer, matches default implementation
complementBit (2147483647) 1 = 2147483645, valid Integer, matches default implementation
complementBit (2147483647) 62 = 4611686020574871551, valid Integer, matches default implementation
complementBit (2147483647) 63 = 9223372039002259455, valid Integer, matches default implementation
complementBit (2147483647) 64 = 18446744075857035263, valid Integer, matches default implementation
complementBit (2147483648) 0 = 2147483649, valid Integer, matches default implementation
complementBit (2147483648) 1 = 2147483650, valid Integer, matches default implementation
complementBit (2147483648) 62 = 4611686020574871552, valid Integer, matches default implementation
complementBit (2147483648) 63 = 9223372039002259456, valid Integer, matches default implementation
complementBit (2147483648) 64 = 18446744075857035264, valid Integer, matches default implementation
complementBit (2147483649) 0 = 2147483648, valid Integer, matches default implementation
complementBit (2147483649) 1 = 2147483651, valid Integer, matches default implementation
complementBit (2147483649) 62 = 4611686020574871553, valid Integer, matches default implementation
complementBit (2147483649) 63 = 9223372039002259457, valid Integer, matches default implementation
complementBit (2147483649) 64 = 18446744075857035265, valid Integer, matches default implementation
complementBit (9223372036854775805) 0 = 9223372036854775804, valid Integer, matches default implementation
complementBit (9223372036854775805) 1 = 9223372036854775807, valid Integer, matches default implementation
complementBit (9223372036854775805) 62 = 4611686018427387901, valid Integer, matches default implementation
complementBit (9223372036854775805) 63 = 18446744073709551613, valid Integer, matches default implementation
complementBit (9223372036854775805) 64 = 27670116110564327421, valid Integer, matches default implementation
complementBit (9223372036854775806) 0 = 9223372036854775807, valid Integer, matches default implementation
complementBit (9223372036854775806) 1 = 9223372036854775804, valid Integer, matches default implementation
complementBit (9223372036854775806) 62 = 4611686018427387902, valid Integer, matches default implementation
complementBit (9223372036854775806) 63 = 18446744073709551614, valid Integer, matches default implementation
complementBit (9223372036854775806) 64 = 27670116110564327422, valid Integer, matches default implementation
complementBit (9223372036854775807) 0 = 9223372036854775806, valid Integer, matches default implementation
complementBit (9223372036854775807) 1 = 9223372036854775805, valid Integer, matches default implementation
complementBit (9223372036854775807) 62 = 4611686018427387903, valid Integer, matches default implementation
complementBit (9223372036854775807) 63 = 18446744073709551615, valid Integer, matches default implementation
complementBit (9223372036854775807) 64 = 27670116110564327423, valid Integer, matches default implementation
complementBit (9223372036854775808) 0 = 9223372036854775809, valid Integer, matches default implementation
complementBit (9223372036854775808) 1 = 9223372036854775810, valid Integer, matches default implementation
complementBit (9223372036854775808) 62 = 13835058055282163712, valid Integer, matches default implementation
complementBit (9223372036854775808) 63 = 0, valid Integer, matches default implementation
complementBit (9223372036854775808) 64 = 27670116110564327424, valid Integer, matches default implementation
complementBit (9223372036854775809) 0 = 9223372036854775808, valid Integer, matches default implementation
complementBit (9223372036854775809) 1 = 9223372036854775811, valid Integer, matches default implementation
complementBit (9223372036854775809) 62 = 13835058055282163713, valid Integer, matches default implementation
complementBit (9223372036854775809) 63 = 1, valid Integer, matches default implementation
complementBit (9223372036854775809) 64 = 27670116110564327425, valid Integer, matches default implementation

